.github/workflows/course-schedule-comment.yml

```diff
@@ -0,0 +1,68 @@
+# Based on https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+#
+# This runs with the full repo permissions, but checks out the upstream branch, not the PR.
+# Do not run arbitrary code from the PR here!
+name: Comment PR with Course Schedule
+on:
+  workflow_run:
+    workflows: ["Generate Course Schedule"]
+    types: [completed]
+
+jobs:
+  upload:
+    runs-on: ubuntu-latest
+    if: >
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v4
+
+      - name: "Setup Rust cache"
+        uses: ./.github/workflows/setup-rust-cache
+
+      - name: "Generate Schedule on upstream branch"
+        run: |
+          mkdir -p ./course-schedule
+          cargo run -p mdbook-course --bin course-schedule > upstream-schedule
+
+      - name: "Download artifact from PR workflow"
```

Comment:
Nit: I think this can be simplified by using [download-artifact](https://github.com/actions/download-artifact) action. When I was implementing my workflow, I wasn't aware of this.

Reply:
I'll give it a try! The linked article seemed to say that wouldn't work, but on a closer reading maybe that's not true. Also, it's quite an old article so maybe it works now :)

Reply:
I haven't tried it, so it might indeed be not working :)

Reply:
Ah, it requires a PAT with read access to the source repository, and won't accept the `repository` input parameter without one -- even if that repository might be public.

Reply:
I am ok with not implementing it as well if it is not easy.

Reply:
(It's possible that the script I've used _also_ won't work with a different, even public, repository -- we'll see!)